src/panel/PanelManager.jsx

```diff
@@ -113,26 +118,21 @@ export default class PanelManager extends React.Component {
   }
 
   updateTopBarReturnButton({ poiFilters = {}, isFromFavorite, poi = {} } = {}) {
-    const topBarReturnButton = document.querySelector('.search_form__return-to-list');
-    const backAction =
-      poiFilters.category || poiFilters.query
-        ? this.backToList
-        : isFromFavorite
-        ? this.backToFavorite
-        : () => {};
-    topBarReturnButton.onclick = e => {
-      backAction(e, poiFilters);
-    };
-
     const topBarHandle = document.querySelector('.top_bar');
-
-    // Show return arrow (on mobile) if user comes from PoI / favorites list
+    const topBarReturnButton = document.querySelector('.search_form__return');
     if (poi.name && (poiFilters.category || poiFilters.query || isFromFavorite)) {
+      const backAction =
+        poiFilters.category || poiFilters.query ? this.backToList : this.backToFavorite;
+      // use the mousedown event so it's triggered before the blur event on the suggest
+      topBarReturnButton.onmousedown = e => {
+        if (this.state.isSuggestOpen) {
+          return;
+        }
+        backAction(e, poiFilters);
+      };
       topBarHandle.classList.add('top_bar--poi-from-list');
-    }
-    // Hide return button when not on a POI anymore
-    else {
-      const topBarHandle = document.querySelector('.top_bar');
+    } else {
+      topBarReturnButton.removeAttribute('onmousedown');
```

Comment:
should we remove the attribute on componentWillUnmount too ?

Reply:
Why not but in practice it will have no effect as this component is never unmounted. Actually it's already full of stuff not properly cleaned, it has no `componentWillUnmount` yet…
I'm not saying it's good, but it will be easier to do it properly when :
 a) the top bar is migrated to React (which I will do at last just after this PR is merged, I'm fed up with waiting)
 b) `PanelManager` is written as a functional component, with explicit side effects (which I'll do after the React Top Bar)